src/main.ml

```diff
@@ -147,12 +147,25 @@ let make_message_switch_server () =
   | `Error (`Msg m) ->
     Lwt.fail_with (Printf.sprintf "Failed to listen on message-switch queue: %s" m)
 
+let get_log_level () =
+  let info = Cmdliner.(Term.info "varstored-guard" ~exits:Term.default_exits) in
+  let log_level =
+    let doc = "Syslog level. E.g. debug, info etc." in
+    Cmdliner.(Term.(const Syslog.level_of_string $ Arg.(value & opt string "info" & info ["log-level"] ~docv:"LEVEL" ~doc))) in
+  Cmdliner.Term.eval (log_level, info)
+
 let () =
-  let old_hook = !Lwt.async_exception_hook in
-  Lwt.async_exception_hook := (fun exn ->
-      D.log_backtrace ();
-      D.error "Lwt caught async exception: %s" (Printexc.to_string exn);
-      old_hook exn
-    );
-  let () = Lwt_main.run @@ make_message_switch_server () in
-  D.debug "Exiting varstored-guard"
+  let main () =
+    let old_hook = !Lwt.async_exception_hook in
+    Lwt.async_exception_hook := (fun exn ->
+        D.log_backtrace ();
+        D.error "Lwt caught async exception: %s" (Printexc.to_string exn);
+        old_hook exn
+      );
+    let () = Lwt_main.run @@ make_message_switch_server () in
+    D.debug "Exiting varstored-guard" in
+
+  Debug.set_facility Syslog.Local5;
+  match get_log_level () with
+  | `Ok log_level -> Debug.set_level log_level; main ()
```

Comment:
Usually we let cmdliner drive the program from here, we don't query individual arguments, like this:
```
Term.exit @@ Term.eval cmd
```